File: src/main/java/de/storchp/opentracks/osmplugin/utils/FileItem.java
package de.storchp.opentracks.osmplugin.utils;

import android.net.Uri;

import androidx.annotation.NonNull;
import androidx.documentfile.provider.DocumentFile;

import java.io.File;

public record FileItem(String name, Uri uri, File file, DocumentFile documentFile) {

    @NonNull
    @Override
    public String toString() {
        return name();
    }

}

File: src/main/java/de/storchp/opentracks/osmplugin/utils/FileItem.kt
package de.storchp.opentracks.osmplugin.utils

import android.net.Uri
import androidx.documentfile.provider.DocumentFile
import java.io.File

data class FileItem(
    val name: String,
    val uri: Uri?,
    val file: File? = null,
    val documentFile: DocumentFile? = null
) {
    override fun toString() = this.name

}
